### DISCUSSION TITLE
Set Role on every query

### DISCUSSION THREAD

**Discussion by zikphil** [sqlalchemy/sqlalchemy#8423] [Usage Questions] [answered]
Hello,

This might be postgresql specific but I am wondering what would be the best way to append SQL to every single queries coming from SQLAlchemy to my postgres server. My use case requires me to assume a certain role, aka: "SET ROLE xxx" before creating/updating objects on the server. Would using a signal do the trick?

Thanks!

**Comment by zzzeek** [comment, marked as answer]
you'd use [before_cursor_execute()](https://docs.sqlalchemy.org/en/14/core/events.html?highlight=before_cursor_execute#sqlalchemy.events.ConnectionEvents.before_cursor_execute) for that

**Reply by zikphil** [reply]
Such quick replies! Much appreciated. :)
